presto-product-tests/src/main/java/com/facebook/presto/tests/cassandra/DataTypes.java

```diff
@@ -0,0 +1,126 @@
+import com.teradata.tempto.Requirement;
+import com.teradata.tempto.RequirementsProvider;
+import com.teradata.tempto.configuration.Configuration;
+import com.teradata.tempto.fulfillment.table.jdbc.RelationalDataSource;
+import com.teradata.tempto.internal.fulfillment.table.cassandra.CassandraTableDefinition;
+import com.teradata.tempto.query.QueryResult;
+import org.testng.annotations.Test;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.net.Inet4Address;
+import java.net.UnknownHostException;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.UUID;
+
+import static com.facebook.presto.tests.TestGroups.CASSANDRA;
+import static com.facebook.presto.tests.cassandra.TestConstants.CONNECTOR_NAME;
+import static com.facebook.presto.tests.cassandra.TestConstants.KEY_SPACE;
+import static com.teradata.tempto.assertions.QueryAssert.Row.row;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.fulfillment.table.TableRequirements.immutableTable;
+import static com.teradata.tempto.query.QueryExecutor.query;
+
+import static java.lang.String.format;
+import static java.sql.JDBCType.BIGINT;
+import static java.sql.JDBCType.BOOLEAN;
+import static java.sql.JDBCType.DOUBLE;
+import static java.sql.JDBCType.INTEGER;
+import static java.sql.JDBCType.LONGNVARCHAR;
+import static java.sql.JDBCType.TIMESTAMP;
+
+public class DataTypes
+        extends ProductTest
+        implements RequirementsProvider
+{
+    private static final String ALL_TYPES_TABLE_NAME = "all_types";
+
+    private static final CassandraTableDefinition ALL_TYPES_TABLE_DEFINITION;
+
+    // Leaving out "bl blob" because I don't know how to represent it in Java
+    private static final String ALL_TYPES_DDL =
+           "CREATE TABLE %NAME% (a ascii, b bigint, bo boolean, d decimal, do double," +
+           "f float, fr frozen<set<int>>, i inet, integer int, l list<int>, m map<text,int>," +
+           "s set<int>, t text, ti timestamp, tu timeuuid, u uuid, v varchar, vari varint," +
+           "PRIMARY KEY (a))";
+
+    static {
+        RelationalDataSource dataSource = () -> {
+            try {
+                return ImmutableList.<List<Object>>of(
+                        ImmutableList.of("abc", 123L, false, new BigDecimal(456.789), 1234.5678,
+                                90123.456f, ImmutableSet.of(1, 2, 3), Inet4Address.getByName("127.0.0.1"), 1, ImmutableList.of(4, 5, 6), ImmutableMap.of("a", 1, "b", 2),
+                                ImmutableSet.of(4, 5, 6), "this is a text value", Timestamp.valueOf("2011-02-03 04:05:00"),
+                                UUID.fromString("d2177dd0-eaa2-11de-a572-001b779c76e3"), UUID.fromString("01234567-0123-0123-0123-0123456789ab"),
+                                "abc", new BigInteger("567")),
+                        ImmutableList.of("def", 123L, false, new BigDecimal(456.789), 1234.5678,
```

Comment:
IMO that's something we should test at a lower level than product tests (e.g. the integration tests should test that).